libevmasm/PeepholeOptimiser.cpp

```diff
@@ -428,6 +428,51 @@ struct TruthyAnd: SimplePeepholeOptimizerMethod<TruthyAnd>
 	}
 };
 
+// (a == true) -> a and (true == a) -> a
+struct TruthyBooleanComparison: SimplePeepholeOptimizerMethod<TruthyBooleanComparison>{
+	static bool applySimple(
+		AssemblyItem const& _pushOrDupN,
+		AssemblyItem const& _iszero,
+		AssemblyItem const& _iszero1,
+		AssemblyItem const& _pushOrDupN1,
+		AssemblyItem const& _iszero2,
+		AssemblyItem const& _iszero3,
+		AssemblyItem const& _subOrEq,
+		std::back_insert_iterator<AssemblyItems> _out
+	)
+	{
+		if (
+			_iszero == Instruction::ISZERO &&
+			_iszero1 == Instruction::ISZERO &&
+			_iszero2 == Instruction::ISZERO &&
+			_iszero3 == Instruction::ISZERO
+		){
+			if (
+				_pushOrDupN.type() == Push && _pushOrDupN.data() == 1 &&
+				SemanticInformation::isDupInstruction(_pushOrDupN1)
+			){
+				*_out = AssemblyItem(dupInstruction(getDupNumber(_pushOrDupN1.instruction()) - 1));
+			} else if (
+				SemanticInformation::isDupInstruction(_pushOrDupN) &&
+				_pushOrDupN1.type() == Push && _pushOrDupN1.data() == 1
+			){
+				*_out = _pushOrDupN;
+			} else
+				return false;
+
+			if (_subOrEq == Instruction::SUB)
+			{
+				*_out = _iszero;
+				return true;
+			} else if (_subOrEq == Instruction::EQ)
+				return true;
+			else
+				return false;
+		} else
+			return false;
```

Comment:
By the way, this is the style we use:

```suggestion
			if (_subOrEq == Instruction::SUB)
			{
				*_out = _iszero;
				return true;
			}
			else if (_subOrEq == Instruction::EQ)
				return true;
			else
				return false;
		} 
		else
			return false;
```